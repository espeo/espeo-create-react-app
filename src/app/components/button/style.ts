import styled from 'styled-components';

interface PropsButton {
  color?: string;
  backgroundColor?: string;
  borderColor?: string;
  width?: string;
  type?: string;
  clickHandler(): void;
}

export const StyledButton = styled.button`
  color: ${(props: PropsButton) => props.color};
  border-color: ${(props: PropsButton) => props.borderColor};
  backgroud-color: ${(props: PropsButton) => props.backgroundColor};
  border-radius: ${(props: any) => (props.shape === 'round' ? '50%' : '3px')};
  width: ${(props: PropsButton) =>
    props.width === 'small' ? '60px' : '100px'};
  height: 20px;
`;
