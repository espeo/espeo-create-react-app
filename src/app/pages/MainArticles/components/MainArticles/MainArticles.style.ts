import { styled } from '@core/styles/themes/defaultTheme';
import { Link } from 'react-router-dom';

export const Select = styled.select`
  height: 20px;
  width: 80px;
  margin-right: 2%;
  border-radius: 4px;
  border-color: ${props => props.theme.colors.darkViolet};
`;

export const ArticlesWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: center;
  margin-top: 3%;
`;

export const ArticleWrapper = styled.div`
  width: 29%;
  padding-right: 2%;
  padding-left: 2%;
  padding-top: 2%;
`;

export const FilterButtonWrapper = styled.div`
  padding-left: 2%;
`;

export const FiltersWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding-top: 2%;
`;

export const FilterWrapper = styled.div`
  display: flex;
  margin-right: 0.5%;
`;

export const ArticleButton = styled(Link)`
  display: block;
  text-decoration: none;
  text-align: center;
  width: ${(props: { width?: string }) => (props.width ? props.width : '100%')};
  background-color: ${props => props.theme.colors.darkViolet};
  border-radius: 5px;
  color: ${props => props.theme.colors.light};
  font-size: 16px;
  line-height: 24px;
`;
