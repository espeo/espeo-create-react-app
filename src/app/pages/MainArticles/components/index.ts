export * from './ArticlesWrapper/ArticlesWrapper';
export * from './ArticleWrapper/ArticleWrapper';
export * from './FiltersWrapper/FiltersWrapper';
export * from './FilterWrapper/FilterWrapper';
export * from './FilterButtonWrapper/FilterButtonWrapper';
export * from './Select/Select';
