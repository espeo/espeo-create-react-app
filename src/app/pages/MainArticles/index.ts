import { connect } from 'react-redux';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';

import MainArticles from './components/MainArticles/MainArticles.component';
import {
  fetchArticles,
  filterArticles,
  clearArticlesFilters,
} from './store/actions';

const mapStateToProps = (state: any): any => {
  const { page, topic, sortBy, date, data } = state.articles;
  return {
    page,
    topic,
    sortBy,
    date,
    articles: data,
  };
};

const mapDispatchToProps: any = {
  fetchArticles,
  filterArticles,
  clearArticlesFilters,
};

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  injectIntl,
)(MainArticles);
