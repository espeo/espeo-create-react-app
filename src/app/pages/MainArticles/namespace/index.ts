import { dateValues } from '../store/saga/index';

export interface MainArticlesState {
  isLoading: boolean;
  isErr: boolean;
  page: number;
  topic: string;
  sortBy: string;
  date: dateValues;
  data: Array<MainArticlesStateData>;
}

export interface MainArticlesStateData {
  author: string;
  content: string;
  description: string;
  publishedAt: string;
  source: {
    id: null | string;
    name: string;
  };
  id: null;
  name: string;
  title: string;
  url: string;
  urlToImage: string;
}
