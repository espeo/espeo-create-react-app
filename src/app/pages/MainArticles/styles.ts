import { styled } from '@styles/themes/defaultTheme';

export const Select = styled.select`
  height: 20px;
  width: 80px;
  margin-right: 2%;
  border-radius: 4px;
  border-color: ${props => props.theme.colors.darkViolet};
`;

export const ArticlesWrapper = styled.div`
  display: flex;
  flex-wrap: wrap;
  flex-direction: center;
  margin-top: 3%;
`;

export const ArticleWrapper = styled.div`
  width: 29%;
  padding-right: 2%;
  padding-left: 2%;
  padding-top: 2%;
`;
export const FilterButtonWrapper = styled.div`
  padding-left: 2%;
`;
export const FiltersWrapper = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  padding-top: 2%;
`;

export const FilterWrapper = styled.div`
  display: flex;
  margin-right: 0.5%;
`;
