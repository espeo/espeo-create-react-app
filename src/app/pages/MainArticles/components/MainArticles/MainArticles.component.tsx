import React, { ChangeEvent, PureComponent } from 'react';
import dayjs from 'dayjs';
import { InjectedIntlProps } from 'react-intl';
import { v1 } from 'uuid';

import { MainTitle, StyledButton } from '@core/components';
import {
  AppWrapper,
  ArticleAuthor,
  ArticleDate,
  ArticleDescription,
  ArticleImage,
  ArticleTitle,
  ContentWrapper,
  TitleWrapper,
} from '@core/styles/components';
import { defaultTheme } from '@core/styles/themes';
import {
  ArticleButton,
  ArticlesWrapper,
  ArticleWrapper,
  FilterButtonWrapper,
  FiltersWrapper,
  FilterWrapper,
  Select,
} from './MainArticles.style';
import { MainArticlesStateData } from '../../namespace';

interface StringObject {
  [key: string]: string | number;
}

interface OwnProps {
  fetchArticles(page: number, filters: StringObject): {};
  filterArticles(filters: StringObject): void;
  clearArticlesFilters(): void;
  articles: Array<MainArticlesStateData>;
  topic: string;
  sortBy: string;
  date: string;
  handleDate(): void;
}
const fallbackImage = require(`@assets/images/logo_espeo.svg`);

type MainArticlesProps = OwnProps & InjectedIntlProps;

class MainArticles extends PureComponent<MainArticlesProps> {
  public componentDidMount() {
    const { fetchArticles } = this.props;
    fetchArticles(1, { topic: '', sortBy: '', date: '' });
  }

  private changeTitle = (title: string) => {
    if (!title) {
      return '';
    }

    return title.toLowerCase().replace(/ /g, '-');
  };

  private handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
    const { filterArticles, topic, sortBy, date } = this.props;
    const { name } = event.target;
    const value = event.target.value === name ? '' : event.target.value;
    if (name === 'topic') {
      filterArticles({ [name]: value, sortBy, date });
    } else if (name === 'sortBy') {
      filterArticles({ [name]: value, topic, date });
    } else if (name === 'date') {
      filterArticles({ [name]: value, topic, sortBy });
    }
  };

  public render() {
    const {
      intl,
      articles,
      topic,
      sortBy,
      date,
      clearArticlesFilters,
    } = this.props;

    return (
      <AppWrapper className="main-page">
        <TitleWrapper>
          <MainTitle text={intl.formatMessage({ id: 'page.main.title' })} />
        </TitleWrapper>
        <ContentWrapper>
          <FiltersWrapper className="filters-wrapper">
            <FilterWrapper className="filters">
              <Select
                value={date}
                onChange={this.handleChange}
                id="select-date"
                name="date"
                className="filter"
              >
                <option value="" />
                <option value="today">
                  {intl.formatMessage({ id: 'mainArticles.filter.today' })}
                </option>
                <option value="week">
                  {intl.formatMessage({ id: 'mainArticles.filter.week' })}
                </option>
                <option value="month">
                  {intl.formatMessage({ id: 'mainArticles.filter.month' })}
                </option>
              </Select>
              <Select
                value={topic}
                onChange={this.handleChange}
                id="select-topic"
                name="topic"
                className="filter"
              >
                <option value="" />
                <option value="sport">
                  {intl.formatMessage({ id: 'mainArticles.topic.sport' })}
                </option>
                <option value="fashion">
                  {intl.formatMessage({ id: 'mainArticles.topic.fashion' })}
                </option>
                <option value="design">
                  {intl.formatMessage({ id: 'mainArticles.topic.design' })}
                </option>
                <option value="literature">
                  {intl.formatMessage({ id: 'mainArticles.topic.literature' })}
                </option>
              </Select>
              <Select
                value={sortBy}
                onChange={this.handleChange}
                id="select-sort"
                name="sortBy"
                className="filter"
              >
                <option value="" />
                <option value="popularity">
                  {intl.formatMessage({ id: 'mainArticles.sortBy.popularity' })}
                </option>
                <option value="writenIn">
                  {intl.formatMessage({ id: 'mainArticles.sortBy.writenIn' })}
                </option>
                <option value="publishedAt">
                  {intl.formatMessage({
                    id: 'mainArticles.sortBy.publishedAt',
                  })}
                </option>
              </Select>
            </FilterWrapper>
            <FilterButtonWrapper className="filter-button">
              <StyledButton
                type="button"
                text={intl.formatMessage({
                  id: 'mainArticles.filter.buttonClear',
                })}
                clickHandler={clearArticlesFilters}
                backgroundColor={defaultTheme.colors.darkViolet}
                borderColor={defaultTheme.colors.darkViolet}
                color={defaultTheme.colors.light}
              />
            </FilterButtonWrapper>
          </FiltersWrapper>
          <ArticlesWrapper className="articles">
            {articles &&
              articles.map((article: MainArticlesStateData) => (
                <ArticleWrapper key={article.author + v1()}>
                  <ArticleImage
                    alt={article.title}
                    src={article.urlToImage ?? fallbackImage}
                  />
                  <ArticleDate>
                    {dayjs(article.publishedAt).format('YYYY, MMM DD ')}
                  </ArticleDate>
                  {article.author ? (
                    <ArticleAuthor>{article.author}</ArticleAuthor>
                  ) : (
                    <ArticleAuthor>
                      {intl.formatMessage({ id: 'mainArticles.author' })}
                    </ArticleAuthor>
                  )}
                  <ArticleTitle>{article.title}</ArticleTitle>
                  <ArticleDescription>{article.description}</ArticleDescription>
                  <ArticleButton
                    to={{
                      pathname: `/details/:${this.changeTitle(article.title)}`,
                      state: { article },
                    }}
                    className="article-button"
                  >
                    {intl.formatMessage({ id: 'mainArticles.knowMore' })}
                  </ArticleButton>
                </ArticleWrapper>
              ))}
          </ArticlesWrapper>
        </ContentWrapper>
      </AppWrapper>
    );
  }
}

export default MainArticles;
