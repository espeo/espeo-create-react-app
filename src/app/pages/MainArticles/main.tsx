import React, { PureComponent, ChangeEvent } from 'react';
import { Link } from 'react-router-dom';
import { InjectedIntlProps } from 'react-intl';
import { v1 } from 'uuid';
import dayjs from 'dayjs';
import { defaultTheme } from '@core/styles/themes';

import {
  AppWrapper,
  ContentWrapper,
  ArticleDate,
  ArticleAuthor,
  ArticleTitle,
  ArticleDescription,
  ArticleImage,
  Title,
  TitleWrapper,
} from '@core/components';
import {
  StyledButton,
  ArticleButton,
  ArticleButtonText,
} from '@core/components/button';
import {
  ArticlesWrapper,
  ArticleWrapper,
  FiltersWrapper,
  FilterWrapper,
  FilterButtonWrapper,
  Select,
} from './components';

interface StringObject {
  [key: string]: string | number;
}

interface OwnProps {
  fetchArticles(arg1: number, arg2: StringObject): {};
  filterArticles(arg: StringObject): void;
  clearArticlesFilters(): void;
  articles: Array<any>;
  topic: string;
  sortBy: string;
  date: string;
  handleDate(): void;
}
const image = require('../../components/image/images/logo_espeo.svg');

type MainArticlesProps = OwnProps & InjectedIntlProps;

class MainArticles extends PureComponent<MainArticlesProps> {
  public componentDidMount() {
    const { fetchArticles } = this.props;
    fetchArticles(1, { topic: '', sortBy: '', date: '' });
  }

  changeTitle = (title: string) => {
    return title.toLowerCase().replace(/ /g, '-');
  };

  handleChange = (event: ChangeEvent<HTMLSelectElement>) => {
    const { filterArticles, topic, sortBy, date } = this.props;
    const { name } = event.target;
    const value = event.target.value === name ? '' : event.target.value;
    if (name === 'topic') {
      filterArticles({ [name]: value, sortBy, date });
    } else if (name === 'sortBy') {
      filterArticles({ [name]: value, topic, date });
    } else if (name === 'date') {
      filterArticles({ [name]: value, topic, sortBy });
    }
  };

  render() {
    const {
      intl,
      articles,
      topic,
      sortBy,
      date,
      clearArticlesFilters,
    } = this.props;

    return (
      <AppWrapper>
        <TitleWrapper>
          <Title text={intl.formatMessage({ id: 'page.main.title' })} />
        </TitleWrapper>
        <ContentWrapper>
          <FiltersWrapper>
            <FilterWrapper>
              <Select
                value={date}
                onChange={this.handleChange}
                id="select-date"
                name="date"
              >
                <option value="" />
                <option value="today">Today</option>
                <option value="week">Last week</option>
                <option value="month">Last month</option>
              </Select>
              <Select
                value={topic}
                onChange={this.handleChange}
                id="select-topic"
                name="topic"
              >
                <option value="" />
                <option value="sport">Sport</option>
                <option value="fashion">Fashion</option>
                <option value="design">Design</option>
                <option value="literature">Literature</option>
              </Select>
              <Select
                value={sortBy}
                onChange={this.handleChange}
                id="select-sort"
                name="sortBy"
              >
                <option value="" />
                <option value="popularity">Popularity</option>
                <option value="writenIn">Written in</option>
                <option value="publishedAt">Published at</option>
              </Select>
            </FilterWrapper>
            <FilterButtonWrapper>
              <StyledButton
                type="button"
                text="Clear"
                clickHandler={clearArticlesFilters}
                backgroundColor={defaultTheme.colors.darkViolet}
                borderColor={defaultTheme.colors.darkViolet}
                color={defaultTheme.colors.light}
              />
            </FilterButtonWrapper>
          </FiltersWrapper>
          <ArticlesWrapper>
            {articles[0] &&
              articles[0].map((article: any) => (
                <ArticleWrapper key={article.author + v1()}>
                  <ArticleImage
                    width="100%"
                    height={230}
                    src={article.urlToImage ? article.urlToImage : image}
                  />
                  <ArticleDate>
                    {dayjs(article.publishedAt).format('YYYY, MMM DD ')}
                  </ArticleDate>
                  {article.author ? (
                    <ArticleAuthor>{article.author}</ArticleAuthor>
                  ) : (
                    <ArticleAuthor>Author</ArticleAuthor>
                  )}
                  <ArticleTitle>{article.title}</ArticleTitle>
                  <ArticleDescription>{article.description}</ArticleDescription>
                  <Link
                    to={{
                      pathname: `/details/:${this.changeTitle(article.title)}`,
                      state: { article },
                    }}
                  >
                    <ArticleButton>
                      <ArticleButtonText>Know more</ArticleButtonText>
                    </ArticleButton>
                  </Link>
                </ArticleWrapper>
              ))}
          </ArticlesWrapper>
        </ContentWrapper>
      </AppWrapper>
    );
  }
}

export default MainArticles;
