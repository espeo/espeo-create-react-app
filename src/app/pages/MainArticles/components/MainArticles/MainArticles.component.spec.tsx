import renderer from 'react-test-renderer';
import React from 'react';
import { compose } from 'redux';
import { shallow, mount } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import createSagaMiddleware from 'redux-saga';

import {
  withProviders,
  withAdditionalProperties,
  articleMock as article,
} from '@core/tests/theme-helpers';

import MainArticles from '@core/pages/MainArticles';
import { ArticleWrapper, ArticlesWrapper } from './MainArticles.style';

const mockStore = configureMockStore([createSagaMiddleware()]);

const fetchArticlesMock = jest
  .fn()
  .mockReturnValue({ data: [article, article, article, article, article] });

const WrappedComponent = compose(
  withProviders,
  withAdditionalProperties({
    fetchArticles: fetchArticlesMock,
    store: {
      articles: fetchArticlesMock(),
    },
  }),
)(MainArticles);

let wrapper: any;

beforeEach(() => {
  const store = mockStore({
    articles: fetchArticlesMock(),
    topic: 'snowboard',
  });

  wrapper = mount(
    <BrowserRouter>
      <WrappedComponent />
    </BrowserRouter>,
  );
});

afterEach(() => {
  jest.clearAllMocks();
});

describe('Articles component test suite', () => {
  it('should match snapshot', () => {
    const component = renderer.create(
      <BrowserRouter>
        <WrappedComponent />
      </BrowserRouter>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should fetch articles', () => {
    expect(wrapper.find(<ArticleWrapper />)).toBeTruthy();
    expect(fetchArticlesMock.mock.calls.length).toBe(1);
  });

  it('should mount component with children', () => {
    const text = 'Time for new information';
    expect(wrapper.find('h1').text()).toEqual(text);
  });

  it('should render articles component', () => {
    expect(wrapper.find('.main-page')).toBeTruthy();
  });

  it('should have these five nodes when mounting', () => {
    expect(wrapper.find(<ArticleWrapper />)).toBeTruthy();
    expect(fetchArticlesMock.mock.calls.length).toBe(1);
    expect(wrapper.find(ArticleWrapper).length).toEqual(5);
  });
});
