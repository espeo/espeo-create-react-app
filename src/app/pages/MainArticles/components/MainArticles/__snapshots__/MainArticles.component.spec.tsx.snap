// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component test suite should match snapshot 1`] = `
<div
  className="sc-ifAKCX gNffHp main-page"
>
  <div
    className="sc-bZQynM iCDVNC"
  >
    <h1
      className="sc-bwzfXH jpDOdg"
    >
      Time for new information
    </h1>
  </div>
  <div
    className="sc-EHOje dsHwTr"
  >
    <div
      className="sc-kGXeez clzERL filters-wrapper"
    >
      <div
        className="sc-kpOJdX bSXoFX filters"
      >
        <select
          className="sc-cSHVUG fmIWdR filter"
          id="select-date"
          name="date"
          onChange={[Function]}
        >
          <option
            value=""
          />
          <option
            value="today"
          >
            Today
          </option>
          <option
            value="week"
          >
            Last week
          </option>
          <option
            value="month"
          >
            Last month
          </option>
        </select>
        <select
          className="sc-cSHVUG fmIWdR filter"
          id="select-topic"
          name="topic"
          onChange={[Function]}
        >
          <option
            value=""
          />
          <option
            value="sport"
          >
            Sport
          </option>
          <option
            value="fashion"
          >
            Fashion
          </option>
          <option
            value="design"
          >
            Design
          </option>
          <option
            value="literature"
          >
            Literature
          </option>
        </select>
        <select
          className="sc-cSHVUG fmIWdR filter"
          id="select-sort"
          name="sortBy"
          onChange={[Function]}
        >
          <option
            value=""
          />
          <option
            value="popularity"
          >
            Popularity
          </option>
          <option
            value="writenIn"
          >
            WritenIn
          </option>
          <option
            value="publishedAt"
          >
            PublishedAt
          </option>
        </select>
      </div>
      <div
        className="sc-kgoBCf bEZFBp filter-button"
      >
        <button
          className="sc-bdVaJa bIzCMS"
          color="#fff"
          onClick={[Function]}
          type="button"
        >
          Clear
        </button>
      </div>
    </div>
    <div
      className="sc-kAzzGY chkwkL articles"
    >
      <div
        className="sc-chPdSV xGLbF"
      >
        <img
          alt="Formel E: Nur Marketing-Event oder ernsthafter Motorsport?"
          className="sc-gzVnrw jVNOTN"
          src="https://media0.faz.net/ppmedia/aktuell/sport/287963728/1.6501020/facebook_teaser_fplus/bekommen-sie-die-kurve-die.jpg"
        />
        <p
          className="sc-gZMcBi fqZlMc"
        >
          2019, Nov 24 
        </p>
        <p
          className="sc-gZMcBi sc-VigVT fTLbYf"
        >
          Michael Wittershagen
        </p>
        <p
          className="sc-fjdhpX gxhidL"
        >
          Formel E: Nur Marketing-Event oder ernsthafter Motorsport?
        </p>
        <p
          className="sc-htoDjs bjXPgk"
        >
          Sorgt die Formel E wirklich für eine saubere Zukunft? Die Gegenwart ist es jedenfalls nicht. Und wieso startet die Rennserie im umstrittenen Saudi-Arabien? Ein widersprüchlicheres Sportereignis ist schwer vorstellbar.
        </p>
        <a
          className="sc-dxgOiQ eLmpSD article-button"
          href="/details/:formel-e:-nur-marketing-event-oder-ernsthafter-motorsport?"
          onClick={[Function]}
        >
          Know more
        </a>
      </div>
      <div
        className="sc-chPdSV xGLbF"
      >
        <img
          alt="Formel E: Nur Marketing-Event oder ernsthafter Motorsport?"
          className="sc-gzVnrw jVNOTN"
          src="https://media0.faz.net/ppmedia/aktuell/sport/287963728/1.6501020/facebook_teaser_fplus/bekommen-sie-die-kurve-die.jpg"
        />
        <p
          className="sc-gZMcBi fqZlMc"
        >
          2019, Nov 24 
        </p>
        <p
          className="sc-gZMcBi sc-VigVT fTLbYf"
        >
          Michael Wittershagen
        </p>
        <p
          className="sc-fjdhpX gxhidL"
        >
          Formel E: Nur Marketing-Event oder ernsthafter Motorsport?
        </p>
        <p
          className="sc-htoDjs bjXPgk"
        >
          Sorgt die Formel E wirklich für eine saubere Zukunft? Die Gegenwart ist es jedenfalls nicht. Und wieso startet die Rennserie im umstrittenen Saudi-Arabien? Ein widersprüchlicheres Sportereignis ist schwer vorstellbar.
        </p>
        <a
          className="sc-dxgOiQ eLmpSD article-button"
          href="/details/:formel-e:-nur-marketing-event-oder-ernsthafter-motorsport?"
          onClick={[Function]}
        >
          Know more
        </a>
      </div>
      <div
        className="sc-chPdSV xGLbF"
      >
        <img
          alt="Formel E: Nur Marketing-Event oder ernsthafter Motorsport?"
          className="sc-gzVnrw jVNOTN"
          src="https://media0.faz.net/ppmedia/aktuell/sport/287963728/1.6501020/facebook_teaser_fplus/bekommen-sie-die-kurve-die.jpg"
        />
        <p
          className="sc-gZMcBi fqZlMc"
        >
          2019, Nov 24 
        </p>
        <p
          className="sc-gZMcBi sc-VigVT fTLbYf"
        >
          Michael Wittershagen
        </p>
        <p
          className="sc-fjdhpX gxhidL"
        >
          Formel E: Nur Marketing-Event oder ernsthafter Motorsport?
        </p>
        <p
          className="sc-htoDjs bjXPgk"
        >
          Sorgt die Formel E wirklich für eine saubere Zukunft? Die Gegenwart ist es jedenfalls nicht. Und wieso startet die Rennserie im umstrittenen Saudi-Arabien? Ein widersprüchlicheres Sportereignis ist schwer vorstellbar.
        </p>
        <a
          className="sc-dxgOiQ eLmpSD article-button"
          href="/details/:formel-e:-nur-marketing-event-oder-ernsthafter-motorsport?"
          onClick={[Function]}
        >
          Know more
        </a>
      </div>
      <div
        className="sc-chPdSV xGLbF"
      >
        <img
          alt="Formel E: Nur Marketing-Event oder ernsthafter Motorsport?"
          className="sc-gzVnrw jVNOTN"
          src="https://media0.faz.net/ppmedia/aktuell/sport/287963728/1.6501020/facebook_teaser_fplus/bekommen-sie-die-kurve-die.jpg"
        />
        <p
          className="sc-gZMcBi fqZlMc"
        >
          2019, Nov 24 
        </p>
        <p
          className="sc-gZMcBi sc-VigVT fTLbYf"
        >
          Michael Wittershagen
        </p>
        <p
          className="sc-fjdhpX gxhidL"
        >
          Formel E: Nur Marketing-Event oder ernsthafter Motorsport?
        </p>
        <p
          className="sc-htoDjs bjXPgk"
        >
          Sorgt die Formel E wirklich für eine saubere Zukunft? Die Gegenwart ist es jedenfalls nicht. Und wieso startet die Rennserie im umstrittenen Saudi-Arabien? Ein widersprüchlicheres Sportereignis ist schwer vorstellbar.
        </p>
        <a
          className="sc-dxgOiQ eLmpSD article-button"
          href="/details/:formel-e:-nur-marketing-event-oder-ernsthafter-motorsport?"
          onClick={[Function]}
        >
          Know more
        </a>
      </div>
      <div
        className="sc-chPdSV xGLbF"
      >
        <img
          alt="Formel E: Nur Marketing-Event oder ernsthafter Motorsport?"
          className="sc-gzVnrw jVNOTN"
          src="https://media0.faz.net/ppmedia/aktuell/sport/287963728/1.6501020/facebook_teaser_fplus/bekommen-sie-die-kurve-die.jpg"
        />
        <p
          className="sc-gZMcBi fqZlMc"
        >
          2019, Nov 24 
        </p>
        <p
          className="sc-gZMcBi sc-VigVT fTLbYf"
        >
          Michael Wittershagen
        </p>
        <p
          className="sc-fjdhpX gxhidL"
        >
          Formel E: Nur Marketing-Event oder ernsthafter Motorsport?
        </p>
        <p
          className="sc-htoDjs bjXPgk"
        >
          Sorgt die Formel E wirklich für eine saubere Zukunft? Die Gegenwart ist es jedenfalls nicht. Und wieso startet die Rennserie im umstrittenen Saudi-Arabien? Ein widersprüchlicheres Sportereignis ist schwer vorstellbar.
        </p>
        <a
          className="sc-dxgOiQ eLmpSD article-button"
          href="/details/:formel-e:-nur-marketing-event-oder-ernsthafter-motorsport?"
          onClick={[Function]}
        >
          Know more
        </a>
      </div>
    </div>
  </div>
</div>
`;
